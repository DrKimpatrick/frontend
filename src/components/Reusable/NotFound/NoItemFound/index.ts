import NoItemFound from './NoItemFound';

export { NoItemFound };
