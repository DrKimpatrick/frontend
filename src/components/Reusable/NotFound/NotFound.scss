.pageNotFound {
  height: 100vh;
  width: 100%;
  left: 0;
  right: 0;
  background-color: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  position: relative;
  z-index: 100;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  .items {
    width: 500px;
    position: relative;
    margin: auto;
    ul {
      display: flex;
      justify-content: center;
      align-items: center;
      flex-direction: column;
    }
    li {
      padding: 20px;
      padding-top: 10px;
      padding-bottom: 10px;
      border-bottom: 2px solid #f0f0f0;
      width: 100%;
      text-align: center;
      color: #b3adad;
      font-size: 18px;
      font-weight: bold;
      &:last-child {
        border-bottom: none;
      }
      .status {
        color: #d10000;
        font-size: 100px;
      }
    }
  }
}

@media screen and (max-width: 768px) {
  .pageNotFound {
    .items {
      width: 90%;
    }
  }
}
