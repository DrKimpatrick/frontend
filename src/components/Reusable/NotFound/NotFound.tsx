import React, { FC } from 'react';
import './NotFound.scss';
import { useHistory } from 'react-router-dom';
import { Block, ArrowBack } from '@material-ui/icons';

interface Props {
  message?: string;
  serverError?: boolean;
  back?: boolean;
}

const NotFound: FC<Props> = props => {
  const { message, serverError, back } = props;

  const history = useHistory();

  return (
    <div className="pageNotFound">
      <div className="items">
        <ul>
          <li>
            <h1 className="status">{serverError ? 505 : 404}</h1>
          </li>
          <li>
            <div className="flex justify-center items-center">
              <Block
                style={{ width: '50px', height: '50px', color: '#d10000' }}
              />
              <div className="mx-2">
                {serverError ? 'Server Error' : 'Not Found'}
              </div>
            </div>
          </li>
          <li>
            {message ? (
              <div style={{ fontSize: 15 }}>{message}</div>
            ) : (
              <div style={{ fontSize: 15 }}>
                Page not found, please click on the button below to go back
              </div>
            )}
          </li>
          <li>
            <div
              className="flex justify-center items-center"
              style={{ fontSize: 15 }}
            >
              <button
                type="button"
                onClick={() => (back ? history.goBack() : history.push('/'))}
              >
                <ArrowBack style={{ width: '25px', height: '25px' }} />
              </button>
              <div className="mx-2">go back</div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  );
};

export default NotFound;
