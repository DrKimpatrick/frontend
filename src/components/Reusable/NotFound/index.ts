import NotFound from './NotFound';

export * from './NoItemFound';
export { NotFound };
