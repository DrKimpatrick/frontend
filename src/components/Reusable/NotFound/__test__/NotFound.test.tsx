import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import ReactDom from 'react-dom';
import NotFound from '../NotFound';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
describe('NotFound', () => {
  afterEach(cleanup);

  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDom.render(
      <Router>
        <NotFound />
      </Router>,
      div
    );
  });

  it('should render without crashing', () => {
    const tree = renderer
      .create(
        <Router>
          <NotFound />
        </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render message and server error status', () => {
    const tree = renderer
      .create(
        <Router>
          <NotFound message="not found" serverError={true} back={true} />
        </Router>
      )
      .toJSON();

    expect(tree).toBeTruthy();
  });
});
