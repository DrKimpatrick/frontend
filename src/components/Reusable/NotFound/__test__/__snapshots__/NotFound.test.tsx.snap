// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound should render without crashing 1`] = `
<div
  className="pageNotFound"
>
  <div
    className="items"
  >
    <ul>
      <li>
        <h1
          className="status"
        >
          404
        </h1>
      </li>
      <li>
        <div
          className="flex justify-center items-center"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "color": "#d10000",
                "height": "50px",
                "width": "50px",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM4 12c0-4.42 3.58-8 8-8 1.85 0 3.55.63 4.9 1.69L5.69 16.9C4.63 15.55 4 13.85 4 12zm8 8c-1.85 0-3.55-.63-4.9-1.69L18.31 7.1C19.37 8.45 20 10.15 20 12c0 4.42-3.58 8-8 8z"
            />
          </svg>
          <div
            className="mx-2"
          >
            Not Found
          </div>
        </div>
      </li>
      <li>
        <div
          style={
            Object {
              "fontSize": 15,
            }
          }
        >
          Page not found, please click on the button below to go back
        </div>
      </li>
      <li>
        <div
          className="flex justify-center items-center"
          style={
            Object {
              "fontSize": 15,
            }
          }
        >
          <button
            onClick={[Function]}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              style={
                Object {
                  "height": "25px",
                  "width": "25px",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </button>
          <div
            className="mx-2"
          >
            go back
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
